daemon/worker.c

```diff
@@ -1029,15 +1045,124 @@ deny_refuse(struct comm_point* c, enum acl_access acl,
 			comm_point_drop_reply(repinfo);
 			return 0; /* discard this */
 		}
-		sldns_buffer_set_limit(c->buffer, LDNS_HEADER_SIZE);
-		sldns_buffer_write_at(c->buffer, 4, 
-			(uint8_t*)"\0\0\0\0\0\0\0\0", 8);
+		log_assert(LDNS_QDCOUNT(sldns_buffer_begin(c->buffer)) == 1);
```

Comment:
Not a bug, but a suggestion.

```suggestion
		/* worker_check_request() above guarantees that the buffer contains at least a header and that qdcount ==  1 */
		log_assert(sldns_buffer_limit(pkt) >= LDNS_HEADER_SIZE && LDNS_QDCOUNT(sldns_buffer_begin(c->buffer)) == 1);
```